apps/meteor/app/livechat/server/api/v1/agent.ts

```diff
@@ -58,3 +60,49 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'livechat/agent.status',
+	{ authRequired: true, validateParams: isPOSTLivechatAgentStatusProps },
+	{
+		async post() {
+			const { status } = this.bodyParams;
+			let { agentId } = this.bodyParams;
+			if (!agentId) {
+				agentId = this.userId;
+			}
+
+			const agent = await Users.findOneAgentById(agentId, {
+				projection: {
+					status: 1,
+					statusLivechat: 1,
+				},
+			});
+			if (!agent) {
+				return API.v1.failure('Agent not found');
+			}
+
+			const newStatus = status || (agent.statusLivechat === 'available' ? 'not-available' : 'available');
+			if (newStatus === agent.statusLivechat) {
+				return API.v1.success();
```

Comment:
Should we return the agentId and the new status? Would be good for testing, since we would know what's happening 👀 wdyt?